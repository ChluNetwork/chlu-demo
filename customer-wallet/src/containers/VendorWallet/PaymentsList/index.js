import React from 'react'
//components
import PaymentItem from './PaymentItem'

const PaymentsList = () => (
  <div className='payments-list'>
    <div className='period'>September 2017</div>
    <PaymentItem />
    <PaymentItem />
    <div className='period'>August 2017</div>
  </div>
)

export default PaymentsList
