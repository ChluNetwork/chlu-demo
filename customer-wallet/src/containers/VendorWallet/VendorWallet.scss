.vendor-wallet {
  padding: 0 10px;
}

.crypto {
  margin: 30px 0 40px;

  >h3{
    margin: 5px 0;
  }
}

.payments-list {
  max-width: 450px;

  .period {
    margin: 20px 0;
  }
}

.payment-item {
  margin-bottom: 40px;

  &-head {
    display: flex;
    justify-content: space-between;
    align-items: center;

    .info {
      display: flex;
      align-items: center;

      .avatar {
        max-width: 50px;
        max-height: 50px;
        margin-right: 10px;
      }
    }
  }

  &-date {
    display: flex;
    justify-content: space-between;
    margin: 10px 0;
  }

  &-rating {
    font-size: 25pt;
  }
}
