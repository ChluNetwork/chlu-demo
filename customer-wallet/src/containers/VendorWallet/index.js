import React, { Component } from 'react'
// utils
import { getExchangeRates } from 'utils/exchangeReq'
// libs
import { fx } from 'money'
// components
import PaymentsList from './PaymentsList'
// styles
import './VendorWallet.css'
// data
import rates from 'fixtures/rates'
import data from './assets/data'

class VendorWallet extends Component {

  state = {
    ratesLoading: true
  }

  componentWillMount () {
    const isTestEnv = process.env.NODE_ENV === 'test'
    isTestEnv ? this.setFxRates(rates) : this.getFxRates()
  }

  getFxRates () {
    getExchangeRates().then((data) => {
      if (data) {
        this.setFxRates(data)
      }

      this.setState({ ratesLoading: false })
    })
  }

  setFxRates (data) {
    fx.rates = data.rates
    fx.base = data.base
  }

  getTotalUsd = (value) => fx.convert(value, { from: 'BTC', to: 'USD' }).toFixed(4)

  render () {
    const { ratesLoading } = this.state
    const totalUsd = !ratesLoading ? this.getTotalUsd(data.total) : null
    return (
      <div className="vendor-wallet">
        <h3>Vendor Wallet</h3>
        {totalUsd &&
          <div className="crypto">
            <h3>{data.total} BTC</h3>
            <span>{totalUsd} USD</span>
          </div>}
        {
          data.reviews.map(({date, reviews}, index) => <PaymentsList date={date} reviews={reviews} key={index}/>)
        }
      </div>
    )
  }
}

export default VendorWallet
