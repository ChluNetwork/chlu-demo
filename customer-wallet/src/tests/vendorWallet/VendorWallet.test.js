import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import VendorWallet from 'containers/VendorWallet'

describe('<VendorWallet />', () => {

  it('should render correctly vendor wallet container', () => {
    const wrapper = shallow(
      <VendorWallet />
    )

    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
