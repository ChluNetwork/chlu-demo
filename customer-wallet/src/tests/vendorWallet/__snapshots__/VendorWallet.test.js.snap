// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<VendorWallet /> should render correctly vendor wallet container 1`] = `
<div
  className="vendor-wallet"
>
  <h3>
    Vendor Wallet
  </h3>
  <PaymentsList
    date="September 2017"
    reviews={
      Array [
        Object {
          "date": "Sept 05, 2017",
          "platform": "Amazon",
          "price": "1.18732 BTC",
          "productPhoto": "no-product.png",
          "productTitle": "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Cras bibendum non turpis et sollicitudin. Aenean ac turpis ac mi molestie faucibus vitae nec diam. In bibendum sem lorem, et faucibus.",
          "rating": 3,
          "review": "Shinny New Item 1",
        },
        Object {
          "date": "Sept 01, 2017",
          "platform": "Exmo",
          "price": "2.12353 BTC",
          "productPhoto": "no-product.png",
          "productTitle": "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Cras bibendum non turpis et sollicitudin. Aenean ac turpis ac mi molestie faucibus vitae nec diam.",
          "rating": 2,
          "review": "Shinny New Item 2",
        },
      ]
    }
  />
  <PaymentsList
    date="August 2017"
    reviews={
      Array [
        Object {
          "date": "Aug 15, 2017",
          "platform": "Amazon",
          "price": "1.231241 BTC",
          "productPhoto": "no-product.png",
          "productTitle": "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Cras bibendum non turpis et sollicitudin. Aenean ac turpis ac mi molestie faucibus vitae nec diam. In bibendum sem lorem, et faucibus. Cras bibendum non turpis et sollicitudin.",
          "rating": 3,
          "review": "Shinny New Item 3",
        },
        Object {
          "date": "Aug 10, 2017",
          "platform": "Exmo",
          "price": "3.45341 BTC",
          "productPhoto": "no-product.png",
          "productTitle": "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Cras bibendum non turpis et sollicitudin. Aenean ac turpis ac mi molestie faucibus vitae nec diam.",
          "rating": 1,
          "review": "Shinny New Item 4",
        },
      ]
    }
  />
</div>
`;
